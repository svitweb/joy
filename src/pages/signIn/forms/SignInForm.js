import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { renderFormInput } from '../../../components/FormFields';
import { signInValidate } from '../../../services/FormValidate.js';
import Button from '../../../components/button/Button';

function SignInForm({ handleSubmit, onSubmit, valid, loading }) {
	const onSubmitForm = ({ email, password }) => onSubmit(email, password);

	return (
		<form className="auth-page-form" onSubmit={handleSubmit(onSubmitForm)}>
			<h1 className="page-title">Login in Playlisted</h1>
			{/*<div className="btn-group social-btn-group wrap">
				<Button
					iconName='icon-social-media instagram'
					className='full-width'
					type='social'
				>
					<div className="text"><span className="hide-xs">auth</span> met Instagram</div>
				</Button>
				<Button
					className='full-width'
					iconName='icon-social-media facebook'
					type='social'
				>
					<div className="text"><span className="hide-xs">auth</span> met Facebook</div>
				</Button>
			</div>*/}
			<Field name="email" type="text" component={renderFormInput} label="E-mail" />
			<Field name="password" type="password" component={renderFormInput} label="Password" />
			<div className="btn-group submit-btn-group right">
				<Button
					onClick={loading ? undefined : handleSubmit(onSubmitForm)}
					type="primary"
					title="Inloggen"
					disabled={!valid}
					loading={loading}
				/>
			</div>
		</form>
	);
}

SignInForm = reduxForm({
	form: 'signInForm',
	validate: signInValidate,
})(SignInForm);

export default SignInForm;
