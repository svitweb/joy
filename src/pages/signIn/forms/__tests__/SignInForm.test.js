import React from 'react';
import {shallow} from 'enzyme';
import SignInForm from "../SignInForm";

describe('SignInForm', () => {
	it('should render correctly', () => {
		const component = shallow(
			<SignInForm/>
		);
		expect(component).toMatchSnapshot();
	});
});
