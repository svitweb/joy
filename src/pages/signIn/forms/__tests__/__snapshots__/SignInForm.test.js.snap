// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditModal should render correctly 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="signInForm"
  forwardedRef={null}
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
  validate={[Function]}
/>
`;
